sources/web/datalab/static/datalab.js

```diff
@@ -267,6 +267,29 @@ function initializeNotebookApplication(ipy, notebook, events, dialog, utils) {
     notebook.prototype.save_notebook = function() {
       originalSave.apply(this, /* check_lastmodified */ [ false ]);
     }
+
+    // A replacement notebook copy function that makes copies in the root if
+    // the source is under datalab/.
+    notebook.prototype.copy_notebook = function() {
+      var that = this;
+      var base_url = this.base_url;
+      var w = window.open('', IPython._target);
+      var parent = utils.url_path_split(this.notebook_path)[0];
+      if (parent.startsWith('datalab/')) {
+        parent = '';
```

Comment:
Not quite. This does the old behavior if the notebook is not under /datalab (so the copy goes in the old directory). If the notebook is under /datalab then the copy is made in the top-level directory.  So /datalab/samples/BigQuery/n.ipynb would go to /n.ipynb

This was based off a request by Dinesh and is the location he suggested. It would be better in general if we had a 'copy to a different (specified) folder' functionality but in the absence of that we have to pick somewhere.